src/client-private.ts

```diff
@@ -41,26 +50,104 @@ function setupStreamingClient (this: PureCloudWebrtcSdk): Promise<void> {
+  await connection.webrtcSessions.refreshIceServers();
+  log.call(this, 'info', 'PureCloud streaming client ready for WebRTC calls');
 }
 
-function startMedia (this: PureCloudWebrtcSdk) {
+export function proxyStreamingClientEvents (this: PureCloudWebrtcSdk) {
+  // webrtc events
+  const on = this._streamingConnection.webrtcSessions.on.bind(this._streamingConnection);
+  on('requestIncomingRtcSession', onPendingSession.bind(this));
+  on('incomingRtcSession', onSession.bind(this));
+  on('rtcSessionError', this.emit.bind(this, 'error'));
+  on('cancelIncomingRtcSession', session => this.emit('cancelPendingSession', session));
+  on('handledIncomingRtcSession', session => this.emit('handledPendingSession', session));
+  on('traceRtcSession', this.emit.bind(this, 'trace'));
+
+  // other events
+  this._streamingConnection.on('error', this.emit.bind(this, 'error'));
+  this._streamingConnection.on('disconnected', () => this.emit('disconnected', 'Streaming API connection disconnected'));
+}
+
+export async function startGuestScreenShare (this: PureCloudWebrtcSdk): Promise<void> {
+  const stream = await startDisplayMedia.call(this);
+  const jid = parseJwt(this._jwt).data.jid;
+  this._streamingConnection.webrtcSessions.initiateRtcSession({
+    stream,
+    jid,
+    mediaPurpose: 'screenshare'
+  });
+  // Should we attach the video stream on the guest's DOM?
+  // attachScreenShareMedia.call(this, stream);
+  temporaryOutboundStream = stream;
+}
+
+function getDefaultChromeConstraints (): MediaStreamConstraints {
+  if (window.navigator.mediaDevices.getDisplayMedia) {
+    return {
+      audio: false,
+      video: {
+        displaySurface: 'monitor'
+      }
+    } as MediaStreamConstraints;
+  }
+  return {
+    audio: false,
+    video: {
+      mandatory: {
+        chromeMediaSource: 'desktop',
+        maxWidth: window.screen.width,
+        maxHeight: window.screen.height,
+        maxFrameRate: 15
+      }
+    }
+  } as MediaStreamConstraints;
+}
+
+function startDisplayMedia (this: PureCloudWebrtcSdk): Promise<MediaStream> {
+  if (window.navigator.getDisplayMedia) {
+    return window.navigator.getDisplayMedia({ video: true });
+  } else if (window.navigator.mediaDevices.getDisplayMedia) {
+    return window.navigator.mediaDevices.getDisplayMedia({ video: true });
+  } else {
+    return window.navigator.mediaDevices.getUserMedia({ video: { mediaSource: 'screen' } as any });
+    // Not sure what to do here. If this is for chrome, should we check the browser first?
+    // return window.navigator.mediaDevices.getUserMedia(getDefaultChromeConstraints());
+  }
+}
+
+function startAudioMedia (this: PureCloudWebrtcSdk): Promise<MediaStream> {
   return this.pendingStream
     ? Promise.resolve(this.pendingStream)
     : window.navigator.mediaDevices.getUserMedia({ audio: true });
 }
 
+function createScreenShareElement (): HTMLVideoElement {
```

Comment:
See next comment down. I don't think we need this at all. This isn't from webchat.js is it?